src/Voron/Data/CompactTrees/CompactKey.cs

```diff
@@ -19,20 +19,29 @@ namespace Voron.Data.CompactTrees;
 public sealed unsafe class CompactKey : IDisposable
```

Comment:
This is something that we create a _lot_ of, no?

Reply:
Yes, and therefore we are very careful to reuse the instances.

Reply:
This make me nervous, because we may be easily leaking those instances leading to a perf cost
Given that the cost of creating this, we at least previously had the context to clear that which we know that would happen

Reply:
We ended up reusing them in managed mode (a long time ago) because they are too costly to create too and reusing them was causing too many issues down the line. This code has been profiled and optimized carefully over the Corax optimization phase.